src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1465,9 +1465,12 @@ private BoundExpression BindFieldExpression(FieldExpressionSyntax node, BindingD
                     }
             }
 
+            // Field will be null when binding a field expression in a speculative
+            // semantic model when the property does not have a backing field.
```

Comment:
It wasn't obvious what to assert here since the `Binder` is not specific to the semantic model (`IsSemanticModelBinder == false`), even when called from that code path.